packages/eslint-config-typescript/react.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * strvcom/eslint-config-typescript
+ *
+ * @author      Danny Kijkov <dankijkov@gmail.com>
+ * @copyright   2021 STRV
+ * @license     http://choosealicense.com/licenses/bsd-3-clause  BSD-3-Clause License
+ */
+
+'use strict'
+
+module.exports = {
+
+  rules: {
+    'react/prop-types': 'off',
+
+    '@typescript-eslint/consistent-type-definitions': 'off',
+
+    '@typescript-eslint/naming-convention': [
+      'warn',
+      {
+        selector: 'memberLike',
+        modifiers: ['private'],
+        format: ['camelCase'],
+        leadingUnderscore: 'forbid',
+      },
+      {
+        selector: 'variable',
+        types: ['boolean'],
+        format: ['PascalCase'],
+        prefix: ['is', 'should', 'has', 'can', 'did', 'will', 'expected'],
+      },
+      {
+        selector: 'typeParameter',
+        format: ['PascalCase'],
+        custom: {
+          regex: 'T[A-Z][a-zA-Z]+$',
```

Comment:
To be honest I forgot, did we in the end agree on prefixing types with `T`? I feel that we talked about stuff like `TKey` and similar in object definitions and how ugly it was but not sure what was the final verdict. 🤔 

Also fixed the regex a bit:
```suggestion
          regex: '^T[A-Z][a-zA-Z]+',
```

Reply:
Afaik we agreed, that for generics (which this rule aims for) we should stick with `T` prefix :) 

I have no rules for `typeProperty` tho, should I add one?

Relevant info: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#allowed-selectors-modifiers-and-types